test/nested_render_test.rb

```diff
@@ -40,7 +43,15 @@ def test_queries_with_auto
     assert_equal [
       'SELECT "projects".* FROM "projects"',
       'SELECT "customers".* FROM "customers" WHERE "customers"."id" IN (?, ?)',
-      'SELECT "widgets".* FROM "widgets" WHERE "widgets"."project_id" IN (?, ?)',
+      'SELECT "widgets".* FROM "widgets" WHERE "widgets"."project_id" IN (?, ?, ?)',
+    ], @queries
+  end
+
+  def test_queries_for_collection_proxies
+    ProjectBlueprint.render(@test_customer.projects.strict_loading, view: :extended_plus_with_widgets)
```

Comment:
I don't think this line isn't testing exactly what we want. When you add the `strict_loading` call, it changes it from `ActiveRecord::Associations::CollectionProxy` to `ActiveRecord::AssociationRelation`, which was one of the "approved" classes prior to this change.

Reply:
Good catch